atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -120,10 +139,13 @@ public CassandraService(
         Supplier<Map<String, String>> hostnamesByIpSupplier =
                 new HostnamesByIpSupplier(this::getAllNonBlacklistedHosts);
         this.hostnameByIpSupplier = Suppliers.memoizeWithExpiration(hostnamesByIpSupplier::get, 1, TimeUnit.MINUTES);
+        this.cassandraClientInstrumentation = cassandraClientInstrumentation;
     }
 
     @Override
-    public void close() {}
+    public void close() {
+        cassandraClientInstrumentation.close();
+    }
```

Comment:
Important as async initialization of the pool objects can fail and be re-tried. We don't want to leak threads. It's generally bad in prod code, and can cause testing things to be slow or take too much memory.